pkg/arm/data/azuredeploy.json

```diff
@@ -210,90 +210,28 @@
         },
         {
             "type": "Microsoft.Storage/storageAccounts",
-            "apiVersion": "2015-06-15",
+            "apiVersion": "2017-10-01",
             "location": "{{ .Manifest.Location }}",
             "name": "{{ .Config.RegistryStorageAccount }}",
-            "properties": {
-                "accountType": "Standard_LRS"
-            }
-        },
-        {
-            "type": "Microsoft.Compute/availabilitySets",
-            "apiVersion": "2016-04-30-preview",
-            "location": "{{ .Manifest.Location }}",
-            "name": "as-compute",
-            "properties": {
-                "managed": true,
-                "platformFaultDomainCount": 2,
-                "platformUpdateDomainCount": 3
+            "sku":{
+                "name": "Standard_LRS"
             }
         },
         {
-            "type": "Microsoft.Network/publicIPAddresses",
-            "apiVersion": "2017-08-01",
-            "copy": {
-                "count": {{ .Manifest.ComputeCount }},
-                "name": "loop"
-            },
+            "type": "Microsoft.Compute/virtualMachineScaleSets",
+            "apiVersion": "2017-12-01",
             "location": "{{ .Manifest.Location }}",
-            "name": "[concat('ip-compute-', copyIndex())]",
-            "properties": {
-                "publicIPAllocationMethod": "Dynamic"
-            },
-            "sku": {
-                "name": "Basic"
-            }
-        },
-        {
-            "type": "Microsoft.Network/networkInterfaces",
-            "apiVersion": "2016-03-30",
-            "copy": {
-                "count": {{ .Manifest.ComputeCount }},
-                "name": "loop"
-            },
             "dependsOn": [
                 "[resourceId('Microsoft.Network/virtualNetworks', 'vnet')]",
                 "[resourceId('Microsoft.Network/networkSecurityGroups', 'nsg-compute')]",
-                "[resourceId('Microsoft.Network/publicIPAddresses', concat('ip-compute-', copyIndex()))]"
+                "[resourceId('Microsoft.Storage/storageAccounts', '{{ .Config.RegistryStorageAccount }}')]"
```

Comment:
init container fails if for some reason all scaleset pool starts before account is created/or failed.

Had a case where account failed to create and all cluster was working but without a registry. I think this would prevent even to start cluster if this is broken. 

Again, we can remove this and keep it as generic deployment error handling.

Reply:
if the account fails to create, that should be caught by checking the arm deployment status.
the init container should retry or crash loop if it can't set up the storage credentials.